src/userobjects/MoabSkinner.C

```diff
@@ -215,9 +222,25 @@ MoabSkinner::getAuxiliaryVariableNumber(const std::string & name,
   return _fe_problem.getAuxiliarySystem().getFieldVariable<Real>(0, name).number();
 }
 
+MooseMesh &
+MoabSkinner::getMooseMesh()
+{
+  if (_use_displaced)
```

Comment:
Can we auto-detect a good default for this? For instance, is `_fe_problem.getDisplacedProblem()` a null pointer if the mesh is not moving? If so, we can use an intelligent default so that users don't make a mistake and forget to enable this when they want it.

Reply:
I added that exactly in line 299 after I kept getting a seg fault while testing which I realized later was because I was `_fe_problem.getDisplacedProblem()->updateMesh();` a non-displaced problem. Do you mean I should do this everywhere as well and remove the `use_displaced_mesh = false` option? I added it this way because I found other solid mechanics module objects use it as an input parameter as well so just following the same logic

Reply:
After thinking about this more I think your design is fine -- let's use the same terminology as in the solid mechanics module for consistency.